perception/autoware_compare_map_segmentation/src/voxel_grid_map_loader/voxel_grid_map_loader.hpp

```diff
@@ -243,7 +245,7 @@ class VoxelGridDynamicMapLoader : public VoxelGridMapLoader
       return;
     }
 
-    (*mutex_ptr_).lock();
+    std::lock_guard<std::mutex> lock(dynamic_map_loader_mutex_);
```

Comment:
If `dynamic_map_loader_mutex_` protects only `current_voxel_grid_dict_`, lock aquisition can be after 
```
current_voxel_grid_array_.assign(
      map_grids_x_ * map_grid_size_y_, std::make_shared<MapGridVoxelInfo>());
```

Reply:
@veqcc Thanks for your comments, I fixed at https://github.com/autowarefoundation/autoware.universe/pull/9097/commits/810003e961234dcbe7bed50b646487b78809c7be#diff-7119dad5db1f62ea22c84fb107a25b505c328d8f2d8e98aaf8ecfb7955f18174